src/gui/accountsettings.cpp

```diff
@@ -339,21 +342,34 @@ void AccountSettings::slotMarkSubfolderEncrypted(FolderStatusModel::SubFolderInf
 
     const auto folder = folderInfo->_folder;
     Q_ASSERT(folder);
+    const auto path = folderInfo->_path;
+    const auto fileId = folderInfo->_fileId;
+    const auto encryptFolder = [this, fileId, folder, path] {
```

Comment:
are you sure folder pointer cannot be made invalid before the lambda is called ?
`QMessageBox::open` is only window modal but maybe the account and folder can be removed via the tray main dialog ?

Reply:
Good catch. It could make sense to pass a folder alias/name and then, request it from the FolderManager within the lambda :)